apps/constellation/main.cpp

```diff
@@ -122,8 +123,25 @@ void InterruptHandler(int /*signal*/)
  * The makes sure that segmentation faults become catchable.
  * This serves as a means to make the VM resiliant to malformed modules.
  */
-void ThrowExeception(int signal)
+namespace {
+std::atomic<bool> shutdown_on_critical_failure{false};
+}
+
+void ThrowException(int signal)
 {
+  FETCH_LOG_ERROR(LOGGING_NAME, "Encountered segmentation fault or floating point exception.");
+  ERROR_BACKTRACE;
+
```

Comment:
Can we make a note to add telemetry for this so that we can monitor these events